### ISSUE TITLE
Locally ping servers in ui serverlist

### ISSUE THREAD

**Issue by Dragonissa** [tiltedphoques/TiltedEvolution#221]
Small QOL to have a ping/latency next to the server in the ui serverlist. Basically what source games have. 

Needs to be discussed which approach makes more sense:

1. Get serverlist in frontend -> send list of ips to core -> core pings and sends back { IP: ping} for each server (json)

or

2. Send request to core -> Get serverlist in core -> send json to frontend -> frontend magic. **(probably better)**

![image](https://user-images.githubusercontent.com/4233773/177131276-a2589617-609b-4d3f-91c7-2f1f49d97c8b.png)

**Comment by danidomen** [comment]
Maybe to send at intervals the ping from the game server side between the game server and list api, to the list api could be? Because also force the list api backend to ping each server could be hard process consuming.

Is like the the proposal of https://github.com/tiltedphoques/TiltedEvolution/issues/247 to send from the game server side the geolocation data.

I know that it will not a real latency between the client and each server, but at least to know which are faster ones. Also I'm not sure that the Valve listing is a latency calculated from the client (is too quickly the results) and are the game servers that are sending at intervals their ping to every geolocated Valve main server (and you get the timings from your Valve/Steam main server region to be more accurate)

**Comment by Dragonisser** [comment]
> Maybe to send at intervals the ping from the game server side between the game server and list api, to the list api could be? Because also force the list api backend to ping each server could be hard process consuming.
> 
> Is like the the proposal of #247 to send from the game server side the geolocation data.
> 
> I know that it will not a real latency between the client and each server, but at least to know which are faster ones. Also I'm not sure that the Valve listing is a latency calculated from the client (is too quickly the results) and are the game servers that are sending at intervals their ping to every geolocated Valve main server (and you get the timings from your Valve/Steam main server region to be more accurate with the latency that you will end to get)

We only have one master server in Europe. Means pining the server in Europe will lead to having a ping of 150+ for America even though the server is 5m next to me.

**Comment by danidomen** [comment]
Maybe a couple of the lower aws ec2 instances (less than 4 bucks/month) to get some NA and AU located instances and get that thing up as Valve style, so each game server send the ping request to them and from that send back to the master server (Europe), and now with the client getting where it is geolocated, then show the result from the list more appropiate to his :)

But, from the client side, if the client need to ping at this moment the 1600 servers while loading the list... well, it could rise a similar to an infite result while never load the list, or doing it in background but not stop the list from loading (and that could make the "Latency" order filter a bit confusing)

Sorry if this comments bother you, I really love all the work and efforts on STR. So anything will be good anyway.

**Comment by Dragonisser** [comment]
> Maybe a couple of the lower aws ec2 instances (less than 4 bucks/month) to get some NA and AU located instances and get that thing up as Valve style, so each game server send the ping request to them and from that send back to the master server (Europe), and now with the client getting where it is geolocated, then show the result from the list more appropiate to his :)
> 
> But, from the client side, if the client need to ping at this moment the 1600 servers while loading the list... well, it could rise a similar to an infite result while never load the list, or doing it in background but not stop the list from loading (and that could make the "Latency" order filter a bit confusing)
> 
> Sorry if this comments bother you, I really love all the work and efforts on STR. So anything will be good anyway.

No, its fine, im just stating things that wont just stating things that wont work out of the box like you've proposed.

Latency doesn't need to be directly available, it can be done in the background, async to not break the server list. Generally having the country will already help in finding a server you would most logically connect to. 

Having multiple services that act as a ping router would probably work, just questionable if we really want to go down that route.
